qiskit/user_config.py

```diff
@@ -138,6 +139,67 @@ def read_config_file(self):
                 self.settings['num_processes'] = num_processes
 
 
+def set_config(key, value, section=None, file=None):
+    """Adds or modifies a user configuration
+
+    It will add the configuration in either the default location
+    ~/.qiskit/settings.conf or the value of file argument.
+
+    Only valid user config can be set in 'default' section. Custom
+    user config can be added in any other sections.
+
+    Args:
+        key (str): name of the config
+        value (str): value of the config
+        section (str, optional): if not specified, adds it to the
+            `default` section of the config file.
+        file (str, optional): the file to which config is added.
+            If not specified, adds it to the default config file.
+
+    Raises:
+        QiskitUserConfigError: if the config is invalid
+    """
+    filename = file or DEFAULT_FILENAME
```

Comment:
Shouldn't this be:

```suggestion
    filename = file or os.getenv('QISKIT_SETTINGS', DEFAULT_FILENAME)
```
so we either write to the specified filename or to the configured settings file.